### ISSUE TITLE
Bug: bdist_wheel fail to build on ubuntu 20.04 fresh install with pyenv 

### ISSUE THREAD

**Issue by sabotagebeats** [ApeWorX/ape#316]
### Environment information

* `ape` and plugin versions:
```
$ ape --version
Traceback (most recent call last):
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (typing-extensions 4.0.1 (/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages), Requirement.parse('typing-extensions<4,>=3.7.4'), {'trie'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/doge/.pyenv/versions/apeworx/bin/ape", line 33, in <module>
    sys.exit(load_entry_point('eth-ape', 'console_scripts', 'ape')())
  File "/home/doge/.pyenv/versions/apeworx/bin/ape", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/doge/ape/ape/src/ape/__init__.py", line 9, in <module>
    from .contracts import _Contract
  File "/home/doge/ape/ape/src/ape/contracts/__init__.py", line 1, in <module>
    from .base import ContractContainer, ContractInstance, ContractLog, _Contract
  File "/home/doge/ape/ape/src/ape/contracts/base.py", line 3, in <module>
    from eth_utils import to_bytes
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/eth_utils/__init__.py", line 4, in <module>
    import pkg_resources
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3252, in <module>
    def _initialize_master_working_set():
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3235, in _call_aside
    f(*args, **kwargs)
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3264, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 585, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 598, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (typing-extensions 4.0.1 (/home/doge/.pyenv/versions/apeworx/lib/python3.8/site-packages), Requirement.parse('typing-extensions<4,>=3.7.4'), {'trie'})
```
$ ape --version
# ...copy and paste result of above command here...

$ ape plugins list
# ...copy and paste result of above command here...

* Python Version: Python 3.8.10
* OS: win 10 ubuntu 20.04 LTS WSL2

### What went wrong?

Please include information like:

* what command you ran
pip install -e . 
* the code that caused the failure (see [this link](https://help.github.com/articles/basic-writing-and-formatting-syntax/) for help with formatting code)
not sure

```Building wheels for collected packages: pytest-watch, ipdb, bitarray, lru-dict, termcolor, docopt, cytoolz, parsimonious, varint, pyethash
  Building wheel for pytest-watch (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/pytest-watch/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/pytest-watch/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-hgmfnsjs
       cwd: /tmp/pip-install-hxf0gtbi/pytest-watch/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for pytest-watch
  Running setup.py clean for pytest-watch
  Building wheel for ipdb (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/ipdb/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/ipdb/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-3ch_qi1c
       cwd: /tmp/pip-install-hxf0gtbi/ipdb/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for ipdb
  Running setup.py clean for ipdb
  Building wheel for bitarray (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/bitarray/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/bitarray/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-3bmob709
       cwd: /tmp/pip-install-hxf0gtbi/bitarray/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for bitarray
  Running setup.py clean for bitarray
  Building wheel for lru-dict (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/lru-dict/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/lru-dict/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-iq0zcfev
       cwd: /tmp/pip-install-hxf0gtbi/lru-dict/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for lru-dict
  Running setup.py clean for lru-dict
  Building wheel for termcolor (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/termcolor/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/termcolor/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-4okb95pw
       cwd: /tmp/pip-install-hxf0gtbi/termcolor/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for termcolor
  Running setup.py clean for termcolor
  Building wheel for docopt (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/docopt/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/docopt/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-4u2kd3cb
       cwd: /tmp/pip-install-hxf0gtbi/docopt/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for docopt
  Running setup.py clean for docopt
  Building wheel for cytoolz (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/cytoolz/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/cytoolz/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-b4h8snrd
       cwd: /tmp/pip-install-hxf0gtbi/cytoolz/
  Complete output (7 lines):
  ALERT: Cython not installed.  Building without Cython.
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for cytoolz
  Running setup.py clean for cytoolz
  Building wheel for parsimonious (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/parsimonious/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/parsimonious/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-rsvus56d
       cwd: /tmp/pip-install-hxf0gtbi/parsimonious/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for parsimonious
  Running setup.py clean for parsimonious
  Building wheel for varint (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/varint/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/varint/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-vv2f4ybd
       cwd: /tmp/pip-install-hxf0gtbi/varint/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for varint
  Running setup.py clean for varint
  Building wheel for pyethash (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/doge/.pyenv/versions/apeworx/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hxf0gtbi/pyethash/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hxf0gtbi/pyethash/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-c0yxzxyb
       cwd: /tmp/pip-install-hxf0gtbi/pyethash/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help
  
  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for pyethash
  Running setup.py clean for pyethash
Failed to build pytest-watch ipdb bitarray lru-dict termcolor docopt cytoolz parsimonious varint pyethash
ERROR: trie 2.0.0a5 has requirement typing-extensions<4,>=3.7.4, but you'll have typing-extensions 4.0.1 which is incompatible.
ERROR: sphinx 3.5.4 has requirement docutils<0.17,>=0.12, but you'll have docutils 0.17.1 which is incompatible.
ERROR: flake8 3.9.2 has requirement pycodestyle<2.8.0,>=2.7.0, but you'll have pycodestyle 2.8.0 which is incompatible.
ERROR: sphinx-rtd-theme 0.5.2 has requirement docutils<0.17, but you'll have docutils 0.17.1 which is incompatible.
```

### How can it be fixed?
unknown

**Comment by fubuloubu** [comment]
Please try running this in bash:

```sh
$ sudo apt install zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev tk-dev libssl-dev llvm libncurses5-dev libncursesw5-dev xz-utils libffi-dev liblzma-dev python-openssl
```

Then remove and re-install the Python version via pyenv e.g.:

```sh
$ pyenv install 3.8.12
```

**Comment by sabotagebeats** [comment]
was able to resolve with the following: `pyenv install 3.9.9 && pyenv virtualenv 3.9.9 myenvironment && pyenv activate myenvironment`

or alternatively with `pip install typing-extensions==3.10.0.2`

the reason this works is because `typing-extensions` is an issue because `eth-tester` has an old pin for `py-evm`
https://github.com/ethereum/eth-tester/blob/master/setup.py#L26

**Comment by sabotagebeats** [comment]
https://github.com/ethereum/eth-tester/pull/229 for tracking

**Comment by sabotagebeats** [comment]
also should revert this once those blockers are resolved https://github.com/ApeWorX/ape/pull/388

**Comment by unparalleled-js** [comment]
That ethereum-tester ticket has been merged and #388 merged.

Is there anything left on this ticket @sabotagebeats ?

**Comment by sabotagebeats** [comment]
I think this is good
